src/Salt/SiteBundle/Controller/CommentsController.php

```diff
@@ -162,30 +162,71 @@ public function downvoteAction(Comment $comment, UserInterface $user)
      */
     public function exportCommentAction(string $itemType, int $itemId)
     {
-        $url = ($itemType === 'item') ? $this->generateUrl('doc_tree_item_view', ['id' => $itemId], UrlGeneratorInterface::ABSOLUTE_URL) : $this->generateUrl('doc_tree_view', ['slug' => $itemId], UrlGeneratorInterface::ABSOLUTE_URL);
+        $childIds = [];
+        $filePointer = fopen('php://temp', 'r+');
         $repo = $this->getDoctrine()->getManager()->getRepository(Comment::class);
-        $headers = ['Framework Name', 'Node Address', ($itemType === 'item') ? 'HumanCodingScheme' : null, 'User', 'Organization', 'Comment', 'Created Date', 'Updated Date'];
-        $rows[] = implode(',', array_filter($headers));
-        $comment_data = $repo->findBy([$itemType => $itemId]);
+        $headers = ['Framework Name', 'Node Address', 'HumanCodingScheme', 'User', 'Organization', 'Comment', 'Created Date', 'Updated Date'];
+        fputcsv($filePointer, $headers);
+        $lsItemRepo = $this->getDoctrine()->getManager()->getRepository(LsItem::class);
+        switch ($itemType)
+        {
+            case 'document':
+                $comment_data = $repo->findBy([$itemType => $itemId]);
+                $this->csvArray($comment_data, $itemType, $filePointer);
+                $lsDoc = $this->getDoctrine()->getManager()->getRepository(LsDoc::class)->find($itemId);
+                $lsDocChilds = $lsDoc->getLsItems();
+                foreach ($lsDocChilds as $lsDocChild){
+                    $childIds[] = $lsDocChild->getId();
+                }
+                break;
+
+            case 'item':
+                $lsItem = $lsItemRepo->findOneById($itemId);
+                $lsItem->getDescendantIds($childIds);
+                $childIds[] = $itemId;
+                break;
+        }
+        $comment_data = $repo->findBy(['item' => $childIds]);
+        $this->csvArray($comment_data, 'item', $filePointer);
+        rewind($filePointer);
+        $csv = stream_get_contents($filePointer);
+        fclose($filePointer);
+        $response = new Response($csv);
```

Comment:
If you move the response creation to the top and use a `StreamedResponse` then you can use a callback function using a `php://output` file in it, not needing to create a (possibly large) string to hold the temporary data.  The `csvArray()` method should return the array (like its name infers) and then you can just do the fputcsv within the callback